src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -445,6 +445,13 @@ def module_is_offloaded(module):
                 f"It seems like you have activated model offloading by calling `enable_model_cpu_offload`, but are now manually moving the pipeline to GPU. It is strongly recommended against doing so as memory gains from offloading are likely to be lost. Offloading automatically takes care of moving the individual components {', '.join(self.components.keys())} to GPU when needed. To make sure offloading works as expected, you should consider moving the pipeline back to CPU: `pipeline.to('cpu')` or removing the move altogether if you use offloading."
             )
 
+        # Enable generic support for intel gaudi accelerator using GPU/HPU migration
+        if device_type == "hpu" and kwargs.pop("hpu_migration", True):
+            os.environ["PT_HPU_GPU_MIGRATION"] = "1"
+            os.environ["PT_HPU_MAX_COMPOUND_OP_SIZE"] = "1"
+
+            import habana_frameworks.torch.core as htcore # noqa: F401
```

Comment:
Do we need this import to be always present?

```
import habana_frameworks.torch.core as htcore
```

?

Also, since we're setting some environment variables here, maybe we could `logger.debug()` this info?

Reply:
@regisss @sayakpaul
Thanks for the quick review!

@sayakpaul 
Yes the import is needed, else we will get error `ModuleNotFoundError: No module named 'torch.hpu'`.  This requirement is also documented [here](https://docs.habana.ai/en/latest/PyTorch/PyTorch_Model_Porting/GPU_Migration_Toolkit/GPU_Migration_Toolkit.html#enabling-the-gpu-migration-toolkit) (see **step 2**).

Good idea to add `logger.debug()`, I will add this next.